crates/router/src/core/payment_methods/cards.rs

```diff
@@ -3167,7 +3167,9 @@ pub async fn list_payment_methods(
         .and_then(|bp| bp.collect_billing_details_from_wallet_connector);
     Ok(services::ApplicationResponse::Json(
         api::PaymentMethodListResponse {
-            redirect_url: merchant_account.return_url,
+            redirect_url: business_profile
```

Comment:
Confirm with respective team before doing this changes.

Reply:
We are not updating the `return_url` merchant account directly, so the updated field will be in `business_profile`. Also the SDK is not using this field